src/main/java/net/helpscout/api/ApiClient.java

```diff
@@ -37,34 +37,32 @@
 public class ApiClient {
 
     private final static String DEFAULT_BASE_URL = "https://api.helpscout.net/v1/";
-    private final static String METHOD_GET = "GET";
-    private final static String METHOD_POST = "POST";
-    private final static String METHOD_PUT = "PUT";
-    private final static String METHOD_DELETE = "DELETE";
 
     private final static int HTTP_STATUS_OK = 200;
     private final static int HTTP_STATUS_CREATED = 201;
 
-    private String apiKey = "";
-    private String baseUrl = DEFAULT_BASE_URL;
-
     private static final ResultExtractor<Long> idExtractor = new IdExtractor();
     private static final ResultExtractor<String> hashExtractor = new HashExtractor();
+    private static final HTTPMethodWrapper httpMethodWrapper = new HTTPMethodWrapper();
 
     private static ApiClient instance = new ApiClient();
 
-    private ApiClient() {}
+    private final JsonParser jsonParser = new JsonParser();
+
+    private ApiClient() {
+        httpMethodWrapper.setBaseUrl(DEFAULT_BASE_URL);
```

Comment:
Hi @bob983 
Sorry that did not answer for a long time. I was on vacation, so I just returned. Actually it is easy change, it can be done for sure. But I'm confused with the idea that service class, will take a care of some specific parameter of system. I mean that URL of api is really belongs to ApiClient class by logic and it is as for me bad move to spread system specific parameter to service class, which should not know anything about system.

One more point - I would not also delete a private contructor of ApiClient, cause we are using here Singelton. So actually it is not a change that can be done for sure from my point of view.